### ISSUE TITLE
ES6 promises not working

### ISSUE THREAD

**Issue by bouncehead13** [johannesjo/angular-promise-buttons#43]
I'm currently using ES6 with my Angular 1.5.X code. The promise button library is failing to resolve the promises because of this code block.
```
mVal.finally(function() {
    promiseDone = true;
    handleLoadingFinished(btnEl);
});
```

`mVal.finally` is not a function with native ES6 promises. I noticed in your angular 2 version, you check for finally and then have a fallback.
```
if (promise.finally) {
    promise.finally(resolveLoadingState);
} else {
    promise
        .then(resolveLoadingState)
        .catch(resolveLoadingState);
}
```

Is it possible to have this approach included into this repo as well.

**Comment by johannesjo** [comment]
Yes definitely!  Thanks for reporting! Would you like to create a PR for this?

**Comment by bouncehead13** [comment]
PR created for you to review #44

**Comment by johannesjo** [comment]
Thank you very much. I will do that!

**Comment by bouncehead13** [comment]
Awesome! I took some time to learn the tests before creating the PR, thought I almost had it, but hit a road block trying to create a native `Promise` in replace of `$q.defer()`.

Definitely available to help if need be.

**Comment by johannesjo** [comment]
I wonder how do you make es 6 promises work for you?

I added unit tests for them in these two commits:
https://github.com/johannesjo/angular-promise-buttons/commit/5f8e5611053ee9d35cbb228d7128aba9159a5814
https://github.com/johannesjo/angular-promise-buttons/commit/af61d1d6437aade0c4b9a6b717ba137a9ddb9a61

But I'm unable to get the unit tests to react to the promise being resolved no matter how many scope.$apply() I'm adding.

**Comment by bouncehead13** [comment]
1) I just pulled the latest released code into my app and it's working great!
2) For the unit tests, the second commit listed, did that not work? Reading the code, it looks right.
